Providers/nxOMSContainers/containers/conf/container.conf

```diff
@@ -0,0 +1,141 @@
+# Fluentd config file for OMS Docker components
+
+# Forward port 25225 for container logs
+<source>
+	type forward
+	port 25225
+	bind 127.0.0.1
+</source>
+
+# Filter container logs
+<filter docker.**>
+	type filter_docker_log
+	log_path "/var/opt/microsoft/omsagent/log/filter_docker_log.txt"
+</filter>
+
+# Container perf
+<source>
+	type oms_omi
+	object_name "Container"
+	interval 15s
+</source>
+
+# Container inventory
+<source>
+	type omi
+	run_interval 60s
+	tag oms.container.containerinventory
+	items [
+		["root/cimv2","Container_ContainerInventory"]
+	]
+</source>
+
+# Image inventory
+<source>
+	type omi
+	run_interval 60s
+	tag oms.container.imageinventory
+	items [
+		["root/cimv2","Container_ImageInventory"]
+	]
+</source>
+
+# Container service log
+<source>
+	type omi
+	run_interval 30s
+	tag oms.container.servicelog
+	items [
+		["root/cimv2","Container_DaemonEvent"]
+	]
+</source>
+
+# Container log
+<source>
+	type omi
+	run_interval 300s
+	tag oms.container.log
+	items [
+		["root/cimv2","Container_ContainerLog"]
+	]
+</source>
+
+# Container host inventory
+<source>
+	type omi
+	run_interval 60s
+	tag oms.api.ContainerNodeInventory
+	items [
+		["root/cimv2","Container_HostInventory"]
+	]
+</source>
+
+# Container process
+<source>
+	type omi
+	run_interval 60s
+	tag oms.api.ContainerProcess
+	items [
+		["root/cimv2","Container_Process"]
+	]
+</source>
+
+#Kubernetes pod inventory
+<source>
+	type kubepodinventory
+	tag oms.api.KubePodInventory
+	run_interval 60s
+</source>
+
+#Kubernetes events
+<source>
+	type kubeevents
+	tag oms.api.KubeEvents
+	run_interval 60s
+</source>
+
+# Filter for correct format to endpoint
+<filter oms.container.**>
+	type filter_container
+</filter>
+
+<match oms.api.KubePodInventory** oms.api.KubeEvents**>
+	type out_oms_api
+	log_level debug
+	buffer_chunk_limit 5m
+	buffer_type file
+	buffer_path /var/opt/microsoft/omsagent/state/out_oms_api_kubernetes*.buffer
+	buffer_queue_limit 10
+	flush_interval 20s
+	retry_limit 10
+	retry_wait 30s
+</match>
+
+
+<match oms.api.ContainerNodeInventory** oms.api.ContainerProcess**>
+  type out_oms_api
+  log_level debug
+  buffer_chunk_limit 20m
+  buffer_type file
+  buffer_path %STATE_DIR_WS%/out_oms_containerinventory*.buffer
```

Comment:
These `%STATE_DIR_WS%` strings should be replaced as the file is placed in the omsagent conf directory.

Reply:
Refer https://github.com/Microsoft/OMS-Agent-for-Linux/blob/ac1c0fe52cb69452b1c606da6eabedcaf2858e88/installer/scripts/omsadmin.sh#L1094

Reply:
The omsadmin.sh script will only replace strings in an explicit set of files at onboarding time. At the time that this resource is placing this conf on the machine, onboarding has already run when this exact file was not placed yet. When this is tested locally, what I expect will be seen is that container.conf is placed in `/etc/opt/microsoft/omsagent/<workspace ID>/conf/omsagent.d/` but container.conf still contains "%STATE_DIR_WS%". This will cause the FluentD engine to fail because it is not a valid path.